--- telepathy-logger-0.1.1/configure.ac	2010-02-26 06:17:43.000000000 +0100
+++ telepathy-logger-0.1.1/configure.ac.new	2010-04-28 09:29:33.733781669 +0200
@@ -19,7 +19,7 @@
 m4_define(tpl_version, tpl_full_version[]tpl_maybe_datestamp)
 
 
-AC_INIT(telepathy-logger, tpl_version, https://bugs.freedesktop.org/buglist.cgi?product=Telepathy&component=logger)
+AC_INIT([telepathy-logger], [tpl_version], [https://bugs.freedesktop.org/buglist.cgi?product=Telepathy&component=logger])
 
 DBUS_SERVICES_DIR="$datadir/dbus-1/services"
 AC_SUBST(DBUS_SERVICES_DIR)
